### PR TITLE
gdalinfo, gdal_translate, gdalwarp: add a -if switch to be able to specify input driver(s)

### PR DESCRIPTION
**Pull request by rouault**
Refs #2641

### CODE REVIEW

File: gdal/doc/source/programs/gdal_translate.rst [hunk 1]
```diff
@@ -51,6 +52,15 @@ resampling, and rescaling pixels in the process.
     Don't be forgiving of mismatches and lost data when translating to the
     output format.
 
+.. option:: -if <FORMAT>
+
+    Format/driver name to be attempted to open the input file. This is generally
+    not necessary, but can be used if automatic driver detection fails to
```
Comment (floppyzedolfin) [inline gdal/doc/source/programs/gdal_translate.rst:58]: Regarding the bug: the automatic detection could be mistaken. `-if` should have precedence over driver checks.

Comment (rouault) [reply gdal/doc/source/programs/gdal_translate.rst:58]: > `-if` should have precedence over driver checks.

that's exactly what it does. Isn't that clear in the documentation ? If not, can you propose some wording ?

Comment (floppyzedolfin) [reply gdal/doc/source/programs/gdal_translate.rst:58]: can be used to skip the automatic driver detection

Comment (rouault) [reply gdal/doc/source/programs/gdal_translate.rst]: I've just reworded it in a slightly different way, hopefully clearer

**Pull request merged**